workers/datasets_based/src/datasets_based/workers/first_rows.py

```diff
@@ -101,6 +103,20 @@ def __init__(self, message: str, cause: Optional[BaseException] = None):
         super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "RowsPostProcessingError", cause, False)
 
 
+class TooManyColumnsError(FirstRowsWorkerError):
+    """Raised when the dataset exceeded the max number of columns."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "TooManyColumnsError", cause, True)
+
+
+class TooBigContentError(FirstRowsWorkerError):
+    """Raised when the first rows content exceeded the max size of bytes."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "TooBigContentError", cause, True)
```

Comment:
It's an internal error (we have not been able to correctly set the environment variables that restrict and truncate the size of the response). The user cannot solve it, it's an issue for us.

```suggestion
        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "TooBigContentError", cause, False)
```